doc: nRF Cloud REST lib: formatting fixes (#9042)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/nrf_cloud_rest.rst

Alternatively, if you have enabled the :kconfig:option:`CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT` option (along with its dependencies), the nRF Cloud REST library generates a JWT automatically if one is not provided.

Socket management and reuse
===========================

The socket used for each HTTP request is determined by the ``connect_socket`` parameter of the :c:struct:`nrf_cloud_rest_context` structure.
If a valid socket file descriptor is passed, this socket is used.
If ``-1`` is passed, a socket will be created automatically.

The selected socket can be reused depending on the passed-in value of the ``keep_alive`` property of the :c:struct:`nrf_cloud_rest_context` structure:

* If you set ``keep_alive`` to false, the selected socket is closed after each request and the ``connect_socket`` property is reset to ``-1``.
* If you set ``keep_alive`` to true, the socket remains open and ``connect_socket`` remains unaltered, meaning the socket is reused on the next API call.

However, if you set ``keep_alive`` to true, make sure that the socket has not been closed externally (for example, due to inactivity) before sending a request.
Otherwise, the request will be dropped.

> A timeout in a socket can manifest in two ways:

Reviewer: ```suggestion
Timeouts in the socket can occur in two ways:
```

> Timeouts in the socket can occur in two ways:

Reviewer 2: ```suggestion
Timeouts in the socket can happen in two ways:
```
Less fancy word, but I always prefer simpler Saxon semantics than words of Latin origin.
